osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/ManagementService.java

```diff
@@ -142,4 +144,14 @@ public void clearMBusStatusOnAllChannels(
     this.clearMBusStatusOnAllChannelsCommandExecutor.execute(
         conn, device, clearMBusStatusOnAllChannelsRequestDto, messageMetadata);
   }
+
+  public UpdateProtocolResponseDto updateProtocol(
+      final DlmsDevice device, final UpdateProtocolRequestDto requestDto) {
+
+    device.setProtocol(requestDto.getProtocol(), requestDto.getProtocolVersion());
```

Comment:
Is it correct that ProtocolVariant is only used in the core, and not in the protocol adapter?

Reply:
Yes, that is correct. The protocol variant field is only added to and relevant in protocol_info tabel in the core database. With this field we can distinguish between different protocols combined with another value (variant). In our case this column will contain the device communication method (CDMA, LTE-M, etc). With this we can configure different protocol adapter queues so we can have different protocol adapter instances depending on the communication method.